addshare/addshare.c

```diff
@@ -33,25 +33,46 @@ enum {
 	COMMAND_UPDATE_SHARE,
 };
 
-static void usage(void)
+static void usage(int status)
 {
 	int i;
 
-	fprintf(stderr, "Usage: smbshareadd\n");
-
-	fprintf(stderr, "\t-a | --add-share=share\n");
-	fprintf(stderr, "\t-d | --del-share=share\n");
-	fprintf(stderr, "\t-u | --update-share=share\n");
-	fprintf(stderr, "\t-o | --options=\"op1=val1 op2=val2...\"\n");
-
-	fprintf(stderr, "\t-c smb.conf\n");
-	fprintf(stderr, "\t-V | --version\n");
-	fprintf(stderr, "\t-v | --verbose\n");
-
-	fprintf(stderr, "Supported share options:\n");
-	for (i = 0; i < KSMBD_SHARE_CONF_MAX; i++)
-		fprintf(stderr, "\t%s\n", KSMBD_SHARE_CONF[i]);
-	exit(EXIT_FAILURE);
+	fprintf(stderr,
+	        "Usage: ksmbd.addshare {-a SHARE | -u SHARE} {-o OPTIONS} [-c SMBCONF] [-v]\n"
+	        "       ksmbd.addshare {-d SHARE} [-c SMBCONF] [-v]\n"
+	        "       ksmbd.addshare {-V | -h}\n");
+
+	if (status != EXIT_SUCCESS)
+		fprintf(stderr, "Try 'ksmbd.addshare --help' for more information.\n");
+	else {
+		fprintf(stderr,
+		        "Configure shares for config file of ksmbd.mountd user mode daemon.\n"
```

Comment:
Can you use tabs instead of whitespaces as indent?

Reply:
@hclee, I indented with tabs and aligned to the opening parenthesis with spaces. Is this not OK?

Reply:
There are warnings from [checkpatch](https://docs.kernel.org/dev-tools/checkpatch.html).
```
ERROR: code indent should use tabs where possible
#100: FILE: addshare/addshare.c:41:
+^I        "Usage: ksmbd.addshare {-a SHARE | -u SHARE} {-o OPTIONS} [-c SMBCONF] [-v]\n"$

ERROR: code indent should use tabs where possible
#101: FILE: addshare/addshare.c:42:
+^I        "       ksmbd.addshare {-d SHARE} [-c SMBCONF] [-v]\n"$

ERROR: code indent should use tabs where possible
#102: FILE: addshare/addshare.c:43:
+^I        "       ksmbd.addshare {-V | -h}\n");$

ERROR: code indent should use tabs where possible
#108: FILE: addshare/addshare.c:49:
+^I^I        "Configure shares for config file of ksmbd.mountd user mode daemon.\n"$
```

it seems this lines can be aligned with tabs.

Reply:
@hclee, `checkpatch` cannot differentiate between indentation and alignment. Aligning with tabs only works if tabsize is 8 (by accident since `fprintf(` is 8 characters). It won't work for any other tabsize. Do you still want me to make the change?

Reply:
checkpatch can differentiate those, but allows tabs and 7 spaces as indentation. we prefer 8 as tabsize according to LInux kernel coding style.
@namjaejeon What do you think about this?

Reply:
Yes, Agreed. I have manually updated the patch to fix the warnings. Thanks for your reivew!